### ISSUE TITLE
New method fitToSection() ? 

### ISSUE THREAD

**Issue by waissbluth** [alvarotrigo/fullPage.js#2575]
When calling moveTo() to change the slide within the same section with `fitToSection: false`, the vertical scroll is not adjusted to make sure that the slide is fully on the screen.

I don't know if this is a desired behavior, but it would be nice to be able to link to a slide within the same section and force the vertical scrolling.

Here is a hack that makes it work:

````
        /**
         * Moves the page to the given section and slide and fits it.
         * Anchors or index positions can be used as params.
         */
        function moveToAndFit(sectionAnchor, slideAnchor){
            isResizing = true;
            var section = getSectionByAnchor(sectionAnchor);

            //do nothing if there's no section with the given anchor name
            if(!section.length) return;

            //default slide
            if (typeof slideAnchor === 'undefined') {
                slideAnchor = 0;
            }

            scrollPage(section, function(){
                scrollSlider(section, slideAnchor);
            });
            isResizing = false;
        }
````

Otherwise, maybe a public `fitCurrentSlide()` method would also be useful, to manually force a slide to the screen.

**Comment by alvarotrigo** [comment]
Why wouldn't it be adjusted already? Are you using `autoScrolling:false`?
I would need a bit more of context. Could you provide an isolated reproduction in jsfiddle or codepen?

**Comment by waissbluth** [comment]
Yes, sorry, i forgot to mention that: `autoScrolling:false`. I'll put toghether a codepen

**Comment by alvarotrigo** [comment]
Why don't you use `fitToSection:true` to start with anyways?

**Comment by waissbluth** [comment]
Here is a codepen: http://codepen.io/waissbluth/pen/wJwXGR

I'm not using `fitToSection: true` for other reasons, but I need that when users navigate with my own buttons (i.e. buttons that call `moveTo()`) the sections are fitted.

**Comment by alvarotrigo** [comment]
An easy trick here:

```javascript
$('button').on('click', function(){
     // Hopefully fit the section we're moving to
     $.fn.fullpage.setFitToSection(false);
     $(document).trigger('scroll');
     $.fn.fullpage.moveTo(2,1);
});
```

**Comment by waissbluth** [comment]
It's not working for me [codepen](http://codepen.io/waissbluth/pen/dvbqNG). Why would `setFitToSection(false)` help?

**Comment by alvarotrigo** [comment]
Another one here with no animation:
```javascript
$('button').on('click', function(){
  // Hopefully fit the section we're moving to
  $.fn.fullpage.setFitToSection(false);
  $.fn.fullpage.silentMoveTo(3);
  $.fn.fullpage.silentMoveTo(2);
  $.fn.fullpage.moveTo(2,1);
});
```

And another probably the ideal one here:
```javascript 
$('button').on('click', function(){
  // Hopefully fit the section we're moving to
  $.fn.fullpage.setFitToSection(false);
  $('.fp-section.active').removeClass('active');
  $.fn.fullpage.moveTo(2,1);
});
```

**Comment by waissbluth** [comment]
Yeah, thanks, I agree that `$('.fp-section.active').removeClass('active');` is the simplest solution. 

Still, it doesn't feel good to mess with the plugin's internal classes; it would be great to have official support for this.

**Comment by alvarotrigo** [comment]
> it would be great to have official support for this.

I'll give it a though!

**Comment by waissbluth** [comment]
Actually, even though the `removeClass('active')` hack [works in the codepen](http://codepen.io/waissbluth/pen/KWPxvx), it does not work in my app, presumably because `destiny == lastScrolledDestiny`. Does that make sense?

**Comment by alvarotrigo** [comment]
What anchors are you using? And how exactly are you calling your `moveTo` method?
It should work if you use the section index instead of the anchor.

**Comment by waissbluth** [comment]
I am using string anchors and yes, it works by switching to the section index.

I am calling it like this: `$.fn.fullpage.moveTo.apply(null, $(this).attr('href').substring(1).split('/'))` where `this` is an `<a>`.

**Comment by waissbluth** [comment]
This is what I ended up with for internal navigation:

````
$("a[href^='#']").click(function() {
  var anchor = $(this).attr('href').substring(1);
  $.fn.fullpage.silentMoveTo(1);
  var ref = anchor.split('/');
  $.fn.fullpage.moveTo(ref[0], ref[1] || 0);
  return false;
});
````

It works well, but it has the hack to force the slides to fit when getting there via link.

**Comment by alvarotrigo** [comment]
Added a new method `fitToSection()` in the dev branch 2.9.4

**Comment by alvarotrigo** [comment]
Available in the latest release 2.9.4
